realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -1209,6 +1211,70 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
         Realm.migrateRealm(config, migration);
     }
 
+    @Test
+    public void renameAndAddField() {
+        RealmMigration migration = new RealmMigration() {
+            @Override
+            public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
+                realm.schema.get("MigrationFieldRenameAndAdd")
+                        .renameField("string1", "string2")
+                        .addField("string1", String.class);
+            }
+        };
+
+        RealmConfiguration config = configFactory.createConfigurationBuilder()
+                .schema(MigrationFieldRenameAndAdd.class)
+                .schemaVersion(2)
+                .migration(migration)
+                .assetFile("rename-and-add.realm")
+                .build();
+
+        Realm realm = Realm.getInstance(config);
+        assertTrue(realm.getSchema().get("MigrationFieldRenameAndAdd").hasField("string1"));
```

Comment:
Perhaps extract this string since it is used 3 times. Doing `className = MigrationFieldRenameAndAdd.class.getSimpleName()` would also make it more resilient towards people refactoring the class.